sound/soc/intel/boards/sof_es8336.c

```diff
@@ -134,15 +105,21 @@ static int sof_8336_trigger(struct snd_pcm_substream *substream, int cmd)
 	case SNDRV_PCM_TRIGGER_START:
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
 	case SNDRV_PCM_TRIGGER_RESUME:
+		if (substream->stream == 0) {
```

Comment:
You don't need delayed work here

Reply:
Currently, there is a problem with the driver: if the machine boots with the headphone connected, it is currently sending the output to the speaker, as the workqueue job is never triggered. The same could also happen if one plugs or unplugs the headphone while the machine is suspended.

Now, I'm not sure if this is the best way to add such logic, or if it should, instead, call gpiod_set_value during probe time and have suspend/resume functions (either there or at es8316) to handle such cases.

Reply:
I think there is an issue here which I saw happening in my case. 
When I play a video in the browser, lets say I pause the video, after 5-6 seconds the trigger callback is called, with the PAUSE_PUSH/SUSPEND/STOP state. This causes both speaker and headphone GPIOs to be deactivated. 

If during the next 5-6 seconds I unpause the video, the trigger callback function is called again with the START/PAUSE_RELEASE/RESUME state which doesn't enable the GPIOs. Also `sof_es8316_speaker_power_event()` will not get called so I have no sound. If I wait more than 10-12 seconds after pausing the video to unpause it then I also get a call to `sof_es8316_speaker_power_event()` which enables the speaker GPIO and I will hear the sound.

It seems to me that the GPIOs should be also handled on the START/PAUSE_RELEASE/RESUME path.